tests/test_sound.py

```diff
@@ -0,0 +1,379 @@
+FS and BLOCKSIZE in autopilot.stim.sound.jackclient.
+
+A TODO is to test the JackClient itself.
+
+Currently only the sound Noise is tested.
+
+These tests cover multiple durations and amplitudes of mono and
+multi-channel Noise, including some edges cases like very short durations
+or zero amplitude.
+
+The rest of this docstring addresses the workaround used to short-circuit
+jackd and JackClient.
+
+Here is the sequence of events that leads to FS and BLOCKSIZE.
+* If an autopilot.core.pilot.Pilot is initialized:
+**  autopilot.core.pilot.Pilot.__init__ checks prefs.AUDIOSERVER,
+    and calls autopilot.core.pilot.Pilot.init_audio.
+**  autopilot.core.pilot.Pilot.init_audio calls 
+    autopilot.external.__init__.start_jackd.
+**  autopilot.external.__init__.start_jackd takes the JACKDSTRING pref 
+    and replaces the token '-rfs' in it with the FS pref. The jackd
+    process is launched and stored in autopilot.external.JACKD_PROCESS.
+    That process may fail or not, we continue anyway.
+**  Next, autopilot.core.pilot.Pilot.init_audio instantiates an
+    autopilot.stim.sound.jackclient.JackClient()
+**  autopilot.stim.sound.jackclient.JackClient.__init__ 
+    initalizes a jack.Client
+**  autopilot.stim.sound.jackclient.JackClient.fs 
+    is set to jack.Client.samplerate. Note that this is either the
+    requested sample rate, or some default value from jack (not Autopilot)
+    if the client did not actually succeed in booting.
+**  autopilot.stim.sound.jackclient.FS (a global variable) is set to 
+    autopilot.stim.sound.jackclient.JackClient.fs
+
+* Later, a sound (e.g., Noise) is initialized.
+**  autopilot.stim.sound.sounds.Noise.__init__ calls super().__init__,
+**  which is autopilot.stim.sound.sounds.Jack_Sound.__init__
+**  autopilot.stim.sound.sounds.Jack_Sound.__init__ 
+    sets `self.fs` to jackclient.FS
+**  autopilot.stim.sound.sounds.Noise.__init__ calls 
+    autopilot.stim.sound.sounds.Noise.init_sound
+**  autopilot.stim.sound.sounds.Noise.init_sound calls 
+    autopilot.stim.sound.sounds.Jack_Sound.get_nsamples
+**  autopilot.stim.sound.sounds.Jack_Sound.get_nsamples 
+    inspects `self.fs`
+
+To remove the dependence on jackd2 and JackClient, the entire first block 
+of code can be circumvented by setting these:
+autopilot.stim.sound.jackclient.FS
+autopilot.stim.sound.jackclient.BLOCKSIZE
+"""
+
+import numpy as np
+import multiprocessing
+import autopilot.prefs
+import autopilot.external
+import autopilot.stim.sound
+
+
+## Ensure we get the same random sound every time
+np.random.seed(0)
+
+
+## Specify needed params to circumvent init_audio
+sample_rate = 192000
+block_size = 1024
+autopilot.stim.sound.jackclient.FS = sample_rate
+autopilot.stim.sound.jackclient.BLOCKSIZE = block_size
+
+# These are only used in Jack_Sound.__del__
+# Setting them here to avoid warnings during garbage collection
+# Why is there one global PLAY and STOP, rather than sound-specific?
```

Comment:
because of the very clumsy way they are piped to the sound server, i agree that it's bad. the server only pulls from one queue, so as written it's not possible to know what sound is feeding it samples. 

one fix could be that chunks are dropped into the queue as `NamedTuple` that indicates the sound ID (also need to finish making unique ids for everything), and then the server grabs from the queue until it's empty/it gets a set of samples from all the sounds. 

another (the one i am moving towards longterm) is for every object to have a networking object that is capable of making different types of connections to different objects -- eg. it starts by having a tcp port bound that is its 'public advertising' port, and then after making initial connection with another object it's possible to make a socket of a different type -- eg. in this case we would probably want to use `ipc` instead of `tcp` which, from what i understand, is as fast or faster than queues. so then all sounds could communicate with the server independently which would also let us clean up a lot of the atrocious logic in the server's process method.

Reply:
thanks! to clarify, I think I wrote this comment really for myself as I was figuring things out, and then forgot to delete it before submitting the PR. It was more of a muse than a suggestion